File: ServiTec/ServiTec/Domain/Models/Categoria.cs
// ============================================================================
// Projecte:      ServiTec - Sistema de Gestió de Restaurants (TFG)
// Autor:         Luca Klein
// Titulació:     Grau en Enginyeria Informàtica (4t Curs)
// Institució:    Universitat de Girona (UdG)
// Fitxer:        Categoria.cs
// Descripció:    Entitat de domini que representa una categoria de productes
//                dins del catàleg del restaurant (ex. Begudes, Entrants, etc.).
// ============================================================================

using System;
using System.Collections.Generic;

namespace ServiTec.Domain.Models;

/// <summary>
/// Entitat que representa la categoria a la qual pertanyen els productes de la carta.
/// </summary>
public partial class Categoria
{
    /// <summary>
    /// Identificador únic de la categoria.
    /// </summary>
    public int IdCategoria { get; set; }

    /// <summary>
    /// Nom de la categoria (ex. "Postres", "Begudes").
    /// </summary>
    public string Nom { get; set; } = null!;

    /// <summary>
    /// Descripció opcional amb informació detallada sobre la categoria.
    /// </summary>
    public string? Descripcio { get; set; }

    /// <summary>
    /// Col·lecció de productes associats a aquesta categoria.
    /// </summary>
    public virtual ICollection<Producte> Productes { get; set; } = new List<Producte>();
}

File: ServiTec_frontend/app/src/main/java/com/example/servitec_frontend/data/model/Categoria.kt
// ============================================================================
// Projecte:      ServiTec - Sistema de Gestió de Restaurants (TFG)
// Autor:         Luca Klein
// Titulació:     Grau en Enginyeria Informàtica (4t Curs)
// Institució:    Universitat de Girona (UdG)
// Fitxer:        Categoria.kt
// Descripció:    Model de dades i DTOs associats a la gestió de categories
//                de la carta del restaurant.
// ============================================================================

package com.example.servitec_frontend.data.model

import com.google.gson.annotations.SerializedName

/**
 * Representa una categoria de productes de la carta del restaurant.
 *
 * @property idCategoria Identificador únic de la categoria.
 * @property nom Nom de la categoria (ex. "Begudes", "Postres").
 * @property descripcio Descripció opcional de la categoria.
 */
data class Categoria(
    @SerializedName("idCategoria") val idCategoria: Int,
    @SerializedName("nom") val nom: String,
    @SerializedName("descripcio") val descripcio: String?
)

/**
 * Objecte de transferència de dades per a l'actualització d'una categoria (PUT).
 *
 * @property nom Nou nom per a la categoria.
 * @property descripcio Nova descripció per a la categoria.
 */
data class UpdateCategoriaDTO(
    @SerializedName("nom") val nom: String,
    @SerializedName("descripcio") val descripcio: String?
)

/**
 * Objecte de transferència de dades per a la creació d'una nova categoria (POST).
 *
 * @property nom Nom de la nova categoria.
 * @property descripcio Descripció opcional de la nova categoria.
 */
data class CreateCategoriaDTO(
    @SerializedName("nom") val nom: String,
    @SerializedName("descripcio") val descripcio: String?
)